### PR TITLE
Rename block visit marker and update visit time to int64

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <benye@amazon.com>

**What this PR does**:

1. Update marker `VisitTime` from `time.Time` to `int64`
2. Rename the marker file name from `block.visit` to `visit-mark.json` to be consistent with other markers from Thanos like `deletion-mark.json`, etc

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/block_visit_marker.go [hunk 1]
```diff
@@ -62,32 +64,35 @@ func ReadBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID stri
...
 }
 
-func UpdateBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID string, compactorID string, blockVisitMarkerWriteFailed prometheus.Counter) error {
+func UpdateBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID string, reader io.Reader, blockVisitMarkerWriteFailed prometheus.Counter) error {
 	blockVisitMarkerFilePath := path.Join(blockID, BlockVisitMarkerFile)
-	blockVisitMarker := BlockVisitMarker{
-		CompactorID: compactorID,
-		VisitTime:   time.Now(),
-	}
-	visitMarkerFileContent, err := json.Marshal(blockVisitMarker)
```
Comment (yeya24) [inline pkg/compactor/block_visit_marker.go:71]: Insead of marshalling the same content multiple times for different blocks, we just pass in a reader

File: pkg/compactor/block_visit_marker.go [hunk 2]
```diff
@@ -18,24 +18,26 @@ import (
 	"github.com/thanos-io/thanos/pkg/block/metadata"
 )
 
-const BlockVisitMarkerFile = "block.visit"
```
Comment (yeya24) [inline pkg/compactor/block_visit_marker.go:21]: Not sure why it is named differently at the beginning, especially the suffix `visit`. Does it intend to be named differently from the Thanos markers?

File: pkg/compactor/block_visit_marker.go [hunk 3]
```diff
@@ -18,24 +18,26 @@ import (
...
 	ErrorBlockVisitMarkerNotFound  = errors.New("block visit marker not found")
 	ErrorUnmarshalBlockVisitMarker = errors.New("unmarshal block visit marker JSON")
 )
 
 type BlockVisitMarker struct {
-	CompactorID string    `json:"compactorID"`
-	VisitTime   time.Time `json:"visitTime"`
+	CompactorID string `json:"compactorID"`
+	// VisitTime is a unix timestamp of when the block was visited (mark updated).
+	VisitTime int64 `json:"visitTime"`
```
Comment (yeya24) [inline pkg/compactor/block_visit_marker.go:32]: I am thinking about adding another `version` field here for possible changes in the future and backwards compatibility.

Comment (alanprot) [reply pkg/compactor/block_visit_marker.go:32]: Yeah.. i like that!

File: pkg/compactor/block_visit_marker.go (hunk 2 above)
Comment (alanprot) [reply pkg/compactor/block_visit_marker.go:21]: I agree.. we should keep in the same pattern..

Maybe `visit-mark.json`?

**Review by alvinlin123 — APPROVED** [review]

**Comment by alexqyle** [comment]
LGTM

File: pkg/compactor/block_visit_marker.go (hunk 3 above)
Comment (alexqyle) [reply pkg/compactor/block_visit_marker.go:32]: Will we introduce version in this PR or separate one?

Comment (yeya24) [reply pkg/compactor/block_visit_marker.go:32]: I can add it to this pr, let me update

File: pkg/compactor/block_visit_marker.go (hunk 2 above)
Comment (yeya24) [reply pkg/compactor/block_visit_marker.go:21]: Yeah updated the name.

**Comment by yeya24** [comment]
PTAL for another review. @alexqyle @alanprot

**Comment by alanprot** [comment]
Nice work. LGTM! thanks

**Review by alanprot — APPROVED** [review]

**Pull request merged**